react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { index: 1 };
+
+    // bind here
+    this.changeIndex = this.changeIndex.bind(this);
+    this.changeIndexArrow = this.changeIndexArrow.bind(this);
+    this.changeIndexCircle = this.changeIndexCircle.bind(this);
+    this.returnImage = this.returnImage.bind(this);
+    this.returnCirclePos = this.returnCirclePos.bind(this);
+  }
+
+  // do not need to bind lifecyle methods
+  componentDidMount() {
+    this.timerId = setInterval(
+      () => this.changeIndex(),
+      1000
+    );
+  }
+
+  // do not need to bind lifecyle methods
+  componentWillUnmount() {
+    clearInterval(this.timerId);
+  }
+
+  changeIndex() {
+    if (this.state.index === this.props.images.length) {
+      this.setState({ index: 1 });
+    } else {
+      this.setState({ index: this.state.index + 1 });
+    }
+  }
+
+  changeIndexArrow(direction) {
+    if (direction === 'prev') {
+      if (this.state.index === 1) {
+        // instead of using constant number this.props.images.length for consistency
+        this.setState({ index: this.props.images.length });
+      } else {
+        this.setState({ index: this.state.index - 1 });
+      }
+    } else {
+      if (this.state.index === this.props.images.length) {
+        this.setState({ index: 1 });
+      } else {
+        this.setState({ index: this.state.index + 1 });
+      }
+    }
+
+    // for smooth transition when carousel is manually changed by user
+    clearInterval(this.timerId);
+    this.timerId = setInterval(
+      () => this.changeIndex(),
+      1000
+    );
```

Comment:
Just a FYI: since you have this code in multiple places you can make a method that handles this